### ISSUE TITLE
Add something about globbing

### ISSUE THREAD

**Issue by macmladen** [HugoGiraudel/sass-guidelines#279]
What is your opinion on globbing?

**PRO:**

* It helps by removing the need to declare everything.
* It is easier to just add file to folder.
* Also, removing file does not require to remember to hunt `@import` line to delete.
* Files should be independent from each other so inclusion does not really matter.

**CON:**

* You cannot influence order of evaluation
* If something is order dependent it can only be influenced by name
* Can produce hard to debug side effects?
* Is not too `libsass`/`node-sass`/`gulp-sass` friendly (but not too hostile either)

**Comment by HugoGiraudel** [comment]
I'm against it simply because CSS is order dependant, but if done right (understand independant modules), it shouldn't matter much and could be used safely. One of our biggest projects at Edenspiekermann (running on Stylus but it doesn't matter) is actually using globbing with 0 issue whatsoever.

Anyway, the guidelines should involve something about globbing for sure.

**Comment by macmladen** [comment]
It is not a rare discussion in many circles on benefits and dangers it could introduce.

And I think this is very common dilemma, to use globbing or not.

I've spotted few problems with `gulp-css-globbing` and `gulp-sass` in their arguments as they may not play nice under certain circumstances and is not easy to debug.

Gulp has certain limitations on how it can perform globbing and when combined, they may introduce unexpected behaviour as they do in case of variables and mxins when combined in imported partials as I tried to solve in https://github.com/dlmanning/gulp-sass/issues/274#issuecomment-145705771
